Add Dockerfile and implement custom location lookup in meshbot.py

davidfries opened the pull request:
- Created a Dockerfile
- Added a new function `get_custom_lookup` in meshbot.py to handle custom location queries for weather information.
- Updated message handling to include custom location lookup in the response options.
- Modified weather_alert_monitor.py to support dynamic alert channel indexing.
- Updated settings.yaml to include configuration options for custom lookup, node shutdown, and alert channel index.

davidfries committed:
Add Dockerfile and implement custom location lookup in meshbot.py

- Created a Dockerfile
- Added a new function `get_custom_lookup` in meshbot.py to handle custom location queries for weather information.
- Updated message handling to include custom location lookup in the response options.
- Modified weather_alert_monitor.py to support dynamic alert channel indexing.
- Updated settings.yaml to include configuration options for custom lookup, node shutdown, and alert channel index.

davidfries commented:
Working on the multi-message bug and refactoring that a bit

davidfries committed:
Refactor split_message function to support pagination and improve message handling in message_listener

davidfries commented:
I'll do another PR this week for doing some lightning/hail stuff. I work for a weather company so I think this can be expanded to really help out those without service but in-range of a mesh running a 'wx node'

davidfries committed:
Merge branch 'main' into fries_features_customloc_docker

oasis6212 commented:
I am slowly processing your pull request. I have found some issues mainly with single message returns being multi message returns under certain circumstances and that was before I lowered the character count to 180. 

I know that 200 sounds reasonable but I have tested that extensively in earlier development and found messages not come through. Even though the meshtastic package didn't error. Thats why I had a very reserved count at 175. Also I believe the lower character count helps when network conditions are less than optimal, which they usually are. 

Im making some changes to correct the single message issues. The implication of custom look up is pretty neat and definitly something thats going in the main branch. 

I do want to get the menu back to a single message too. Which I realize is gonna take some concessions ultimately to make happen.

oasis6212 committed:
Update meshbot.py

Made some changes to remove pagination from single message returns to keep them to a single message. 
Fixed advertise command to send to all instead of requester.
Added "alert" command to menu.

oasis6212 commented:
I made some simple changes to the pull request. I could not reproduce the issues I seen with the 200 character limit before so I kept it. After much toying with the menu I gave up on trying to keep it to a single message. 

I did find one issue. When I used your custom lookup with the coordinates 38.7652/-112.084 I would get a unable to fetch weather data error. Not sure why. It did work with some other locations I tried. 

Let me know what you think @davidfries

davidfries commented:
Your changes LGTM. Let me noodle on those coordinates not looking up properly;that is very strange.

oasis6212 commented:
Well I just tried it again..... and its working fine!? Maybe I had triggered a throttle from the NWS api? Not sure but I think were good to go here. I'm going to update the main. Thanks for your help its much appreciated! @davidfries
